src/messenger/messenger_logic.cpp

```diff
@@ -0,0 +1,34 @@
+#include "messenger/messenger_logic.h"
+
+#include "common/json.h"
+#include "loggers/messenger_logger.h"
+#include "messenger/messenger.h"
+
+namespace terrier::messenger {
+
+void MessengerLogic::ProcessMessage(std::string_view sender, std::string_view message) {
```

Comment:
How are we handling different connection's logic (Like the model's vs the replica's)? 
Will the `MessengerLogic` have some sort of pointer to the logic of them, and invoke based on sender here? 

Will it be better if the MessengerLogic expose a function to register user-specific logic (model's, replica's) during the runtime?

Reply:
Replaced with callback-per-message functionality.